### ISSUE TITLE
twisted.flow has been Pending Deprecation for 2 years.  We should deprecate it already.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3016]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)| @dreid reported|
|-|-|
|Trac ID|trac#3016|
|Type|task|
|Created|2008-01-29 18:29:32Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-flow-3016|

It hasn't been touched in 2 years except for fixing some unit tests 1 year ago, probably as part of fixing bugs in trial.

<details><summary>Searchable metadata</summary>

```
trac-id__3016 3016
type__task task
reporter__dreid dreid
priority__highest highest
milestone__ 
branch__branches_deprecate_flow_3016 branches/deprecate-flow-3016
branch_author__ 
status__closed closed
resolution__fixed fixed
component__flow flow
keywords__ 
time__1201631372000000 1201631372000000
changetime__1201644516000000 1201644516000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid set owner to @dreid<br>@dreid set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_2"></a>|@dreid commented|
|-|-|

(In [22414]) Branching to 'deprecate-flow-3016'

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_3"></a>|@dreid commented|
|-|-|

(In [22415]) Change PendingDeprecationWarning to an actual DeprecationWarning, so we can eventually delete it.
Refs [#3016](https://github.com/twisted/twisted/issues/3016)

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid removed owner<br>@dreid set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @dreid|
|-|-|

This is probably a useful things to have. I won't hurt with asking you further enhancement on this, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_6"></a>|@dreid set status to `closed`|
|-|-|

(In [22417]) Merge deprecate-flow-3016: Update flow's status to deprecated vs pending deprecation.

Replace the current PendingDeprecationWarning (which has been in place for 2 
years) with a DeprecationWarning.  Soon we might actually be able to delete it.

Author: dreid
Reviewer: therve
Fixes [#3016](https://github.com/twisted/twisted/issues/3016)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
